packages/devtools_app/lib/src/screens/logging/logging_screen_v2/logging_table_v2.dart

```diff
@@ -253,19 +262,21 @@ class _LoggingSettingsDialogV2State extends State<LoggingSettingsDialogV2> {
           ),
           const StructuredErrorsToggle(),
           const SizedBox(height: defaultSpacing),
-          _RententionLimitSetting(
-            theme: theme,
-            onRetentionLimitChange: (newRetentionLimit) =>
-                retentionLimit = newRetentionLimit,
+          PositiveIntegerSetting(
+            title: preferences.logging.retentionLimitTitle,
+            subTitle: 'Used to limit the number of log messages retained.',
+            notifier: newRetentionLimit,
           ),
         ],
       ),
       actions: [
         DialogApplyButton(
           onPressed: () {
-            if (retentionLimit != null) {
+            if (newRetentionLimit.value !=
+                preferences.logging.retentionLimit.value) {
               // Save the new retention limit to preferences.
```

Comment:
no need to do this check. You can just unconditionally set 
`preferences.logging.retentionLimit.value = newRetentionLimit.value` because ValueNotifier already handles the equality check.